pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -1046,4 +1051,9 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard {
     function _denormalizeWeight(uint256 weight) private view returns (uint256) {
         return weight.mulUp(_denormWeightSum);
     }
+
+    // converts from normalized form to the internal representation (summing to customDenormWeightSum)
+    function _denormalizeWeight(uint256 weight, uint256 customDenormWeightSum) private pure returns (uint256) {
+        return weight.mulUp(customDenormWeightSum);
+    }
```

Comment:
This may be overkill and we can inline this if you think readability wouldn't be impacted too much.